### DISCUSSION TITLE
Issue with Mac Plus with Radius Accelerator

### DISCUSSION THREAD

**Discussion by Deton154** [BlueSCSI/BlueSCSI-v2#133] [Help]
I have been trying to get a BlueSCSI V2 working with my Mac Plus and ran into a strange issue. My Mac Plus dose have a Radius Accelerator 16 with the 68881 math coprocessor. I mention this because I think this is what is causing the issue. I am able to get the Blue SCSI to work if use firmware BlueSCSI_Pico_2023-10-09_dbac2022.uf2. If I try to use any newer firmware after that build the Mac Plus doesn't see the SCSI drive. Interestingly when I boot with the Blue SCSI with a newer firmware the splash screen for the accelerator isn't displayed as the computer is booting up. This is why I am thinking the accelerator is causing an issue. I am still able to boot off floppies but not the Blue SCSI. To get the Blue SCSI to work with the older firmware this is what I need to have in the ini file:
[SCSI]
Quirks=1 ; Apple
SelectionDelay=0
EnableSelLatch=1 ; On

I noticed that in the new version of the ini maker it says to just use 
[SCSI]
System=MacPlus

I have tried this and multiple combinations of the ini file to see if I could get it to work but didn't have any luck. I did go ahead and add the debugging to the ini file. Here is the log:
[log.txt](https://github.com/BlueSCSI/BlueSCSI-v2/files/14828934/log.txt)

I have also tried a different SD card and different images from ones that I have made to ones downloaded off the links in the WIKI. 

Let me know if anyone has any thoughts why after that firmware my BlueSCSI stops. I was hoping to get a newer firmware on it to play with the WiFi DaynaPORT.

**Comment by erichelgeson** [comment]
Thanks for the debug log - It looks like the bus is just getting garbage - all 0xFF... I suppose the accelerator is not trivial to remove (so dont do that yet)

Could you try with no ini file? Maybe the ROM's in the accelerator fixed the issues the Mac Plus had with SCSI.

Your saying between v2023-10-09 and v2023.11.16 the issue happens? Thats good because there are only 2 changes to the firmware in that release :) I could send you a test firmware with each one of those changes removed for you to try if removing the ini doesnt resolve the issue.

**Reply by Deton154** [reply]
I just tried with no ini file and no luck. There was a slight change in that the accelerator splash screen came back as the computer was booting. As soon as I add the ini with the Plus option it goes away. I then thought maybe I should try to add the other ini options which I needed with the old firmware. So I tried these in the ini file:
SelectionDelay=0
EnableSelLatch=1 ; On

Both options in the file and then one at a time. No change. 

I have had this BlueSCSI for a while but just now started to really dig into it as I wanted try out the WiFi. So I had been using the firmware BlueSCSI_Pico_2023-10-09_dbac2022.uf2 for a long time. That is how I knew it worked with this old of a firmware. So I imagine the change that is breaking this for me is right after this firmware.

**Reply by erichelgeson** [reply]
Can you explicitly test v2023.11.16? (and provide a debug log of that test)

**Reply by Deton154** [reply]
I just gave firmware v2023.11.16 a test. The computer didn't boot. I used the ini file that is working with the older firmware. I have attached the log. 
[log.txt](https://github.com/BlueSCSI/BlueSCSI-v2/files/14856513/log.txt)
At the blinking question mark I did boot the machine with a floppy to see if a drive would show up on the desktop. No luck. I then rebooted the machine to see if that did anything and still the computer didn't see the BlueSCSI. 

I also tried it with out an ini file with this firmware and no change there. 

Just to make sure I wasn't going crazy I did reload the 2023-10-09 firmware and it worked. I rebooted the machine a few times to make sure and every time it booted up.

**Reply by erichelgeson** [reply]
Awesome, that really narrows it down. I'm pulling down my mac plus tonight or tomorrow and hopefully I can reproduce the issue. Please ping me if it's been a few days without an update!

**Reply by Deton154** [reply]
Hey Eric, I wanted to check in to see if you had a chance to pull down your Mac Plus to take a look at this issue I am seeing? Thanks, Rich

**Reply by erichelgeson** [reply]
Thanks for the reminder - and yes I was able to pull the Plus down and did some testing. After debugging I see that Debug has been broke on the Plus for a while. Not something we notice as we don't normally do debugging on a Plus. I'm guessing the timings are just so tight for the Plus that debug is enough to throw everything off. Which unfortunately makes our job here much harder. Also my SCSI analyzer seems to be on the fritz (from 1989) - so me gathering more info to fix this is at a stand still till I can fix that. 

One thing we can do is I'll provide two uf2 files for you to test. Each with one of the changes reverted that was added between the october and novembe release. If that fixes it we'll know atleast what change caused the issue. If not we'll have to wait.

**Reply by Deton154** [reply]
I am good with waiting until you are able to fix your analyzer. I am also willing to test any firmware revisions that you would want and report back my findings. Thanks for all of the assistance.

**Reply by erichelgeson** [reply]
Could you try something for me? Load the old firmware on, backup your drive image, boot up, install the [silver lining drivers](http://macintoshgarden.org/apps/lacie-silverlining-531-more). Update to the latest firmware and see if you can successfully boot.

**Reply by Deton154** [reply]
So I figured out something by accident when trying to load the Silverlining Drivers onto my image. After getting the SilverLining driver installed and then updating the firmware I accidently forgot to plug in the USB power to the BlueSCSI.  With out thinking I unplugged the BlueSCSI from the Macintosh while it was on; then connected the USB power and plugged the BlueSCSI back into the Mac. The Mac booted right away... So I rebooted the computer and it didn't see the drive again. So I unplugged the USB power cable and then plugged it back in and the computer booted. So I then tried my backup image with out the Silverlining Drivers and it worked by power cycling the BlueSCSI while the MAC is at the blinking question mark. So now it appears that with the newer firmware my MacPlus isn't seeing the BlueSCSI on the bus with out a power cycle after boot. I used the v2024.04.02 firmware. I tried two images on the flash card and both images were mounted this way.

**Reply by erichelgeson** [reply]
Ok great, getting somewhere - so when the bluescsi is powered on _after_ the mac starts up, it's OK. I wonder if the accelerator is doing something to the lines, eg, asserting them all so we're getting the junk on the mac. You are still using the `System=MacPlus` right? Could you try without and report? (everything else the same, on 2024.04.02)

Edit: Whoops, that one wont work make a difference, still digging in.

**Reply by Deton154** [reply]
Just to confirm this is the settings I have been testing with in my ini file:
[SCSI]
System=MacPlus
SelectionDelay=0
EnableUnitAttention=0
EnableSelLatch=1 ; On

If I power cycle the BlueSCSI after booting up to the blinking question mark everything seems to work normally. If I try to reboot the computer hangs at a black screen. If I have the BlueSCSI plug in at boot the accelerator splash screen isn't shown, however if I reboot the BlueSCSI the computer will continue to boot from the BlueSCSI. 

So just for fun I just tried again with out an ini file. Now with the BlueSCSI plugged in the computer booted up and showed the accelerator splash screen however the Mac appeared not to see a drive. I power cycled the BlueSCSI and it booted. So I tested this a couple times and it booted this way every time and worked. I played a game for about 15 mins and then shut it down. I didn't notice any issues.
